x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/query/create_query_alert_type.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import { validateNonExact } from '@kbn/securitysolution-io-ts-utils';
+
+import { PersistenceServices, RuleDataClient } from '../../../../../../rule_registry/server';
+import { QUERY_ALERT_TYPE_ID } from '../../../../../common/constants';
+import { ExperimentalFeatures } from '../../../../../common/experimental_features';
+import { ConfigType } from '../../../../config';
+import { SetupPlugins } from '../../../../plugin';
+
+import { queryRuleParams, QueryRuleParams } from '../../schemas/rule_schemas';
+import { queryExecutor } from '../../signals/executors/query';
+
+import { createSecurityRuleTypeFactory } from '../create_security_rule_type_factory';
+
+export const createQueryAlertType = (createOptions: {
+  experimentalFeatures: ExperimentalFeatures;
+  indexAlias: string;
+  lists: SetupPlugins['lists'];
+  logger: Logger;
+  mergeStrategy: ConfigType['alertMergeStrategy'];
+  ruleDataClient: RuleDataClient;
+  version: string;
+}) => {
+  const {
+    experimentalFeatures,
+    indexAlias,
+    lists,
+    logger,
+    mergeStrategy,
+    ruleDataClient,
+    version,
+  } = createOptions;
+  const createSecurityRuleType = createSecurityRuleTypeFactory({
+    indexAlias,
+    lists,
+    logger,
+    mergeStrategy,
+    ruleDataClient,
+  });
+  return createSecurityRuleType<QueryRuleParams, {}, PersistenceServices, {}>({
+    id: QUERY_ALERT_TYPE_ID,
+    name: 'Custom Query Rule',
+    validate: {
+      params: {
+        validate: (object: unknown) => {
+          const [validated, errors] = validateNonExact(object, queryRuleParams);
+          if (errors != null) {
```

Comment:
it looks like the error throwing part of this function will likely be copied in the other rules, so we could consider making a mini-util for that piece